core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystem.java

```diff
@@ -62,6 +64,14 @@ public interface RelDataTypeSystem {
   /** Returns the maximum precision of a NUMERIC or DECIMAL type. */
   int getMaxNumericPrecision();
 
+  /** Returns the minimum scale of a NUMERIC or DECIMAL type. */
+  int getMinNumericScale();
```

Comment:
We currently have both `getMaxScale(SqlTypeName)` and `getMaxNumericScale()`. (The latter delegates to the former.)

So should we instead be creating `getMinScale(SqlTypeName)`? It would allow a type system to have different min scales for (say) a decimal and a binary floating point.

Reply:
I checked the SQL TypeName#getMinScale and it processed the Interval type, which returns -1 for all other types. Do we need to add this method?

Reply:
@julianhyde

Reply:
We should be explicit in the javadoc.

`getMaxScale(SqlTypeName)` seems to return `RelDataType.SCALE_NOT_SPECIFIED` (Integer.MIN_VALUE) for any data type that does not support scale. Can you amend its javadoc to state this?

The javadoc for `getMinScale(SqlTypeName)` should state its behavior for types that do not support scale. I propose `Integer.MAX_VALUE`.

The javadoc for `getMinScale(SqlTypeName)` should also say that the default type system returns 0 for `DECIMAL` and interval types. I know it's an interface, but interfaces can and should say what the default or typical behavior is.

I guess you should change `supportsNegativeScale()` to `supportsNegativeScale(SqlTypeName)`, and remove `getMinNumericScale()`.